tests/integration/operatorhub/cmd_link_test.go

```diff
@@ -0,0 +1,143 @@
+package integration
+
+import (
+	"fmt"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/openshift/odo/tests/helper"
+)
+
+var _ = Describe("odo link command tests for OperatorHub", func() {
+
+	var commonVar helper.CommonVar
+	var oc helper.OcRunner
+
+	BeforeEach(func() {
+		commonVar = helper.CommonBeforeEach()
+		helper.Chdir(commonVar.Context)
+		oc = helper.NewOcRunner("oc")
+	})
+
+	AfterEach(func() {
+		helper.CommonAfterEach(commonVar)
+	})
+
+	Context("Operators are installed in the cluster", func() {
+
+		var etcdOperator string
+		var etcdCluster string
+
+		BeforeEach(func() {
+			// wait till odo can see that all operators installed by setup script in the namespace
+			odoArgs := []string{"catalog", "list", "services"}
+			operators := []string{"etcdoperator", "service-binding-operator"}
+			for _, operator := range operators {
+				helper.WaitForCmdOut("odo", odoArgs, 5, true, func(output string) bool {
+					return strings.Contains(output, operator)
+				})
+			}
+
+			list := helper.Cmd("odo", "catalog", "list", "services").ShouldPass().Out()
+			etcdOperator = regexp.MustCompile(`etcdoperator\.*[a-z][0-9]\.[0-9]\.[0-9]-clusterwide`).FindString(list)
+			etcdCluster = fmt.Sprintf("%s/EtcdCluster", etcdOperator)
+		})
+
+		AfterEach(func() {
+			helper.DeleteProject(commonVar.Project)
+		})
+
+		When("a component and a service are deployed", func() {
+
+			var componentName string
+			var svcFullName string
+
+			BeforeEach(func() {
+				helper.CopyExample(filepath.Join("source", "nodejs"), commonVar.Context)
+				componentName = "cmp-" + helper.RandString(6)
+				helper.Cmd("odo", "create", "nodejs", componentName).ShouldPass()
+
+				serviceName := "service-" + helper.RandString(6)
+				svcFullName = strings.Join([]string{"EtcdCluster", serviceName}, "/")
+				helper.Cmd("odo", "service", "create", etcdCluster, serviceName, "--project", commonVar.Project).ShouldPass()
+
+				helper.Cmd("odo", "push").ShouldPass()
+				oc.PodsShouldBeRunning(commonVar.Project, componentName+`-app-.[a-z0-9-]*`)
```

Comment:
There is already a function called `GetRunningPodNameByComponent` in the CliRunner interface. Can we use it here so as to run the tests on a kubernetes cluster?

Reply:
`GetRunningPodNameByComponent` is not really checking that the pod is running, only existing, what `PodsShouldBeRunning` is doing.

It would be interesting to extend `GetRunningPodNameByComponent` with a `phase` parameter to make it select pods with this specific `status.phase`.